frontend/src/pages/shipments/DownloadCSVMenu.tsx

```diff
@@ -0,0 +1,130 @@
+import { useAuth0 } from '@auth0/auth0-react'
+import format from 'date-fns/format'
+import { FunctionComponent, useMemo } from 'react'
+import Button from '../../components/Button'
+import PlainModal from '../../components/modal/PlainModal'
+import useModalState from '../../hooks/useModalState'
+import {
+  ShipmentQuery,
+  useExportShipmentToCsvMutation,
+} from '../../types/api-types'
+import { formatShipmentName } from '../../utils/format'
+
+interface Props {
+  shipment: ShipmentQuery['shipment']
+}
+
+const DownloadCSVMenu: FunctionComponent<Props> = ({ shipment }) => {
+  const { getAccessTokenSilently } = useAuth0()
+  const [modalIsVisible, showModal, hideModal] = useModalState()
+
+  const [
+    exportShipment,
+    { loading: exportIsProcessing },
+  ] = useExportShipmentToCsvMutation()
+
+  const exportToCSV = async () => {
+    const shipmentExport = await exportShipment({
+      variables: { shipmentId: shipment.id },
+    })
+    const downloadPath = shipmentExport.data?.exportShipment.downloadPath
+
+    if (!downloadPath) {
+      throw new Error("Unable to download the shipment's data")
+    }
+
+    await downloadShipment(downloadPath)
+  }
+
+  const downloadShipment = async (downloadPath: string) => {
+    const accessToken = await getAccessTokenSilently()
+
+    const spreadsheet = await fetch(downloadPath, {
+      headers: { Authorization: `Bearer ${accessToken}` },
+    }).then((res) => res.text())
+
+    // Download the CSV file by creating a link and clicking it. Yay HTML
+    const anchorElement = document.createElement('a')
+    const fileName = `${formatShipmentName(shipment)}.csv`
+
+    if (URL && 'download' in anchorElement) {
+      anchorElement.href = URL.createObjectURL(
+        new Blob([spreadsheet], {
+          type: 'application/octet-stream',
+        }),
+      )
+      anchorElement.setAttribute('download', fileName)
+      document.body.appendChild(anchorElement)
+      anchorElement.click()
+      document.body.removeChild(anchorElement)
+    } else {
+      window.location.href =
+        'data:application/octet-stream,' + encodeURIComponent(spreadsheet)
+    }
```

Comment:
Question (blocking): is this necessary? could use the `/shipment-exports/:id` endpoint instead?

Reply:
If it does need to be done this way, please extract the logic for triggering the download (creating a an anchor el, etc) client-side to a function.

Reply:
The tricky part is that I need to send an auth header to that endpoint, which I don't think can be done if I use the `window.location` API?

Reply:
Okay I refactored this by pointing the client to the endpoint directly, which feels less gross.